api/hypershift/v1beta1/azure.go

```diff
@@ -636,3 +679,43 @@ type AzureKMSKey struct {
 	// +required
 	KeyVersion string `json:"keyVersion"`
 }
+
+// azureAuthenticationConfiguration is a discriminated union type that contains the Azure authentication configuration
+// for a Hosted Cluster. This configuration is used to determine how the Hosted Cluster authenticates with Azure's API,
+// either with managed identities or workload identities.
+//
+// +kubebuilder:validation:XValidation:rule="self.azureAuthenticationConfigType == 'ManagedIdentities' ? has(self.managedIdentities) : !has(self.managedIdentities)", message="managedIdentities is required when azureAuthenticationConfigType is ManagedIdentities, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="self.azureAuthenticationConfigType == 'WorkloadIdentities' ? has(self.workloadIdentities) : !has(self.workloadIdentities)", message="workloadIdentities is required when azureAuthenticationConfigType is WorkloadIdentities, and forbidden otherwise"
+// +union
+type AzureAuthenticationConfiguration struct {
+	// azureAuthenticationConfigType is the type of identity configuration used in the Hosted Cluster. This field is
+	// used to determine which identity configuration is being used. Valid values are "ManagedIdentities" and
+	// "WorkloadIdentities".
+	//
+	// "ManagedIdentities" means that the Hosted Cluster is using managed identities to authenticate with Azure's API.
+	// This is only valid for managed Azure, also known as ARO HCP.
+	//
+	// "WorkloadIdentities" means that the Hosted Cluster is using workload identities to authenticate with Azure's API.
+	// This is only valid for self-managed Azure.
+	//
+	// unionDiscriminator
+	// +kubebuilder:validation:Enum=ManagedIdentities;WorkloadIdentities
```

Comment:
Generally good practice to create a string type alias with constants for the possible values, it's useful for users of the API to be able to compare to the constants